### ISSUE TITLE
Clarify behavior / docs re Get-Content -Raw -Encoding Byte vs. Get-Content -ReadCount 0 -Encoding Byte

### ISSUE THREAD

**Issue by mklement0** [PowerShell/PowerShell#3932]
According to the documentation, the purpose of `Get-Content -Raw` is to [return a _string_](https://github.com/PowerShell/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/Providers/FileSystem-Provider/Get-Content-for-FileSystem.md):

  > Ignores newline characters and returns the entire contents of a file in one string.

By contrast, the purpose of `-Encoding Byte` is to return a _byte array_, without any attempt to interpret the bytes as a _string_ (characters).

As currently documented:

* The two parameters contradict each other.

* The documented way to get the _entire file contents_ as a _single byte array_ is to use `-ReadCount 0` with `-Encoding Byte`:

> When reading from and writing to binary files, use a value of Byte for the Encoding dynamic parameter and a value of 0 for the ReadCount parameter. A ReadCount value of 0 reads the entire file in a single read operation and converts it into a single object (PSObject).

As an aside: note that `PSObject` is incorrect, because a `[System.Object[]]` array is output, which does _not_ derive from `PSObject`.

In practice, as of beta.2:

* `Get-Content -Raw -Encoding Byte` returns a `[System.Byte[]]` array as a _single_ object.

* `Get-Content -ReadCount 0 -Encoding Byte` returns a `[System.Object[]]` array as a _single_ object.

Note that `[System.Byte[]]` is actually the more efficient way to return the bytes of the file as a single object, so, arguably, that's what `Get-Content -ReadCount 0 -Encoding Byte` should output (possibly in addition to `Get-Content -Raw -Encoding Byte`, if the docs are updated accordingly).

Steps to reproduce
------------------

```powershell
'hi' > t.txt

# Based on the current docs, -Raw combined with -Encoding Byte makes no sense.
# Yet, it works, and returns an efficient, strongly typed [System.Byte[]] array.
Get-Content t.txt -Raw -Encoding Byte | Measure-Object | % Count
(Get-Content t.txt -Raw -Encoding Byte).GetType().Name

'---'

# Based on the current docs, using -ReadCount 0 with -Encoding Byte is the way
# to return all bytes at once, as a single array.
# Returns a [System.Object[]] array.
Get-Content t.txt -ReadCount 0 -Encoding Byte | Measure-Object | % Count
(Get-Content t.txt -ReadCount 0 -Encoding Byte).GetType().Name
```

Actual behavior
---------------

```powershell
1
Byte[]
---
1
Object[]
```

Environment data
----------------

```powershell
PowerShell Core v6.0.0-beta (v6.0.0-beta.2)
```

**Comment by iSazonov** [comment]
I believe if conclusion in #3911 is "By design" we should follow our code behavior.

**Comment by iSazonov** [comment]
@mklement0 Please update the Issue.

**Comment by mklement0** [comment]
@iSazonov, I've updated the issue to reflect the parameter name change from `-Encoding Byte` to `-AsByteStream`, corrected the docs link, and I've generally made it clearer what the issue is about.

I don't know how this issue relates to #3911, however.

At this point, let me try to reach a conclusion:

* The docs should be updated to mention combining `-Raw` and `-AsByteStream` as a meaningful combination that outputs a `[byte[]]` array as a single object.

* In the interest of backward compatibility and because the difference will rarely matter, it's fine to have `-ReadCount 0 -AsByteStream` issue an `[object[]]` array instead.

If you agree, we can close this issue.

**Comment by iSazonov** [comment]
> If you agree, we can close this issue.

I agree but we need to get MSFT team confirmation that it is _by design_ and we can set the label. /cc @SteveL-MSFT 
After that we can continue thinking about #3911.
